File: Property_management_system/src/com/property/management/Lease.java
package com.property.management;

import java.util.ArrayList;
import java.util.List;

public class Lease {
    private String leaseID;
    private String propertyID;
    private String tenantID;
    private String startDate;
    private String endDate;
    private double rentAmount;
    public static List<Lease> leases = new ArrayList<>();

    public Lease(String leaseID, String propertyID, String tenantID, String startDate, String endDate, double rentAmount) {
        this.leaseID = leaseID;
        this.propertyID = propertyID;
        this.tenantID = tenantID;
        this.startDate = startDate;
        this.endDate = endDate;
        this.rentAmount = rentAmount;
    }

    // Getters and Setters
    public String getLeaseID() {
        return leaseID;
    }

    public void setLeaseID(String leaseID) {
        this.leaseID = leaseID;
    }

    public String getPropertyID() {
        return propertyID;
    }

    public void setPropertyID(String propertyID) {
        this.propertyID = propertyID;
    }

    public String getTenantID() {
        return tenantID;
    }

    public void setTenantID(String tenantID) {
        this.tenantID = tenantID;
    }

    public String getStartDate() {
        return startDate;
    }

    public void setStartDate(String startDate) {
        this.startDate = startDate;
    }

    public String getEndDate() {
        return endDate;
    }

    public void setEndDate(String endDate) {
        this.endDate = endDate;
    }

    public double getRentAmount() {
        return rentAmount;
    }

    public void setRentAmount(double rentAmount) {
        this.rentAmount = rentAmount;
    }

    // Methods
    public static void createLease(Lease lease) {
        leases.add(lease);
        System.out.println("Lease created: " + lease.leaseID);
    }

    public static void editLease(String leaseID, String newPropertyID, String newTenantID, String newStartDate, String newEndDate, double newRentAmount) {
        for (Lease lease : leases) {
            if (lease.leaseID.equals(leaseID)) {
                lease.setPropertyID(newPropertyID);
                lease.setTenantID(newTenantID);
                lease.setStartDate(newStartDate);
                lease.setEndDate(newEndDate);
                lease.setRentAmount(newRentAmount);
                System.out.println("Lease updated: " + lease.leaseID);
                return;
            }
        }
        System.out.println("Lease not found!");
    }

    public static void terminateLease(String leaseID) {
        leases.removeIf(lease -> lease.leaseID.equals(leaseID));
        System.out.println("Lease terminated: " + leaseID);
    }
}

File: Property_Management_Systems/src/lease.py
class Lease:
    leases = []

    def __init__(self, lease_id, property_id, tenant_id, start_date, end_date, rent_amount):
        self.lease_id = lease_id
        self.property_id = property_id
        self.tenant_id = tenant_id
        self.start_date = start_date
        self.end_date = end_date
        self.rent_amount = rent_amount

    @classmethod
    def create_lease(cls, lease):
        cls.leases.append(lease)
        print(f"Lease {lease.lease_id} created.")

    @classmethod
    def edit_lease(cls, lease_id, new_property_id, new_tenant_id, new_start_date, new_end_date, new_rent_amount):
        for lease in cls.leases:
            if lease.lease_id == lease_id:
                lease.property_id = new_property_id
                lease.tenant_id = new_tenant_id
                lease.start_date = new_start_date
                lease.end_date = new_end_date
                lease.rent_amount = new_rent_amount
                print(f"Lease {lease_id} updated.")
                return
        print("Lease not found!")

    @classmethod
    def terminate_lease(cls, lease_id):
        cls.leases = [
            lease for lease in cls.leases if lease.lease_id != lease_id]
        print(f"Lease {lease_id} terminated.")

    @classmethod
    def view_leases(cls):
        for lease in cls.leases:
            print(f"ID: {lease.lease_id}, Property ID: {lease.property_id}, Tenant ID: {lease.tenant_id}, Start: {lease.start_date}, End: {lease.end_date}, Rent: {lease.rent_amount}")


# Input
lease1 = Lease("L001", "P001", "T001", "2024-01-01", "2025-01-01", 1200.0)
Lease.create_lease(lease1)
Lease.view_leases()

Lease.edit_lease("L001", "P002", "T002", "2024-01-01", "2025-01-15", 1250.0)
Lease.view_leases()

Lease.terminate_lease("L001")
Lease.view_leases()
